File: OOP Lab/Lab 01/Quadratic.java
import java.util.*;
public class Quadratic {
	public static void main(String []args) {
		Scanner sc = new Scanner (System.in);
		double a, b, c, d, x1, x2;
		
		System.out.print("This program will compute the roots of the quadratic equation:\n\t\tax\u00B2 + bx + c = 0\n\n\tEnter 'a': ");
		a = sc.nextDouble();
		System.out.print("\tEnter 'b': ");
		b = sc.nextDouble();
		System.out.print("\tEnter 'c': ");
		c = sc.nextDouble();
		
		d = b*b - 4*a*c;
		
		// '0' for equal, '1' for distinct, and '2' for imaginary
		int roots = (d == 0)?0:((d>0)?1:2);
		switch (roots) {
			case 0: x1 = -b/(2*a);
				System.out.println("\nThe roots of \'"+a+"x\u00B2 + "+b+"x + "+c+" = 0\' are real and equal.\n\tRoots = "+x1+".");
				break;
			case 1: d = Math.sqrt(d);
				x1 = (-b+d)/(2*a);
				x2 = (-b-d)/(2*a);
				System.out.println("\nThe roots of \'"+a+"x\u00B2 + "+b+"x + "+c+" = 0\' are real and distinct.\n\tRoots = "+x1+", "+x2+".");
				break;
			case 2: d = Math.sqrt(Math.abs(d));
				x1 = -b/(2*a);
				x2 = d/(2*a);
				System.out.println("\nThe roots of \'"+a+"x\u00B2 + "+b+"x + "+c+" = 0\' are complex conjugates.\n\tRoots = \t"+x1+" + "+x2+"i\n\t\t\t"+x1+" - "+x2+"i.");
				break;
			default:
				break;
		}
	}	
}

File: DS Lab/Basic Programs/Quadratic.c
// Quadratic.c
// Created by @avikantz on 08/11/15

#include <stdio.h>
#include <math.h>

void getData(float *a, float *b, float *c) {
	printf("\n\tEnter 'a': ");
	scanf("%f", a);
	printf("\tEnter 'b': ");
	scanf("%f", b);
	printf("\tEnter 'c': ");
	scanf("%f", c);
	printf("\n\n");
}

void print_roots(double d, double *x1, double *x2) {
	if (d < 0) {
		printf("\n\tImaginary Roots:\n\t\t%.4f + %.4fi\n\t\t%.4f - %.4fi\n\n", *x1, *x2, *x1, *x2);
	}
	else if (d > 0) {
		printf("\n\tReal and Distinct Roots:\n\t\t%.4f\n\t\t%.4f", *x1, *x2);
	}
	else {
		printf("\n\tReal and Equal Roots:\n\t\t%.4f\n\t\t%.4f", *x1, *x2);
	}
	printf("\n\n");
}

void quadratic(float a, float b, float c, double *x1, double *x2) {
	double d = b*b - 4*a*c;
	if (d < 0) {
		*x1 = -b/(2*a);
		*x2 = sqrt(fabs(d))/(2*a);
		print_roots(d, x1, x2);
	}
	else if (d > 0) {
		*x1 = (-b + sqrt(d))/(2*a);
		*x2 = (-b - sqrt(d))/(2*a);
		print_roots(d, x1, x2);
	}
	else {
		*x1 = *x2 = -b/(2*a);
		print_roots(d, x1, x2);
	}
}

int main(int argc, const char *argv[]) {
	float a, b, c;
	printf("\n\tThis program will calculate roots of a quadratic equation.\n");
	getData(&a, &b, &c);
	double x1, x2;
	quadratic(a, b, c, &x1, &x2);
	return 0;
}
